src/librustc/ty/util.rs

```diff
@@ -578,6 +578,15 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
             bug!("empty_substs_for_def_id: {:?} has type parameters", item_def_id)
         })
     }
+
+    pub fn const_usize(&self, val: usize) -> ConstInt {
```

Comment:
I see, it's probably ok then (still, it'd be nice to prevent misuses of this function)